cosmoz-data-nav.js

```diff
@@ -749,29 +754,31 @@
 						return idx;
 					}
 
-					const incomplete = this.isIncompleteFn(item);
-					this._toggleInstance(element, idx, incomplete);
-
-					if (incomplete) {
-						// no data for item, instance has been toggled
-						// drop task from queue
+					if (this.isIncompleteFn(item)) {
+						element.item = false;
+						// no data for item drop task from queue
 						return;
 					}
 
+					element.__incomplete._showHideChildren(true);
+
 					if (element.item === item) {
-						// already rendered
+						renderRun = idx === this.selected && this._notifyElementResize();
+						// try to resize
 						// drop task from queue
 						return;
 					}
 
-					if (renderRun) {
-						// one render per run
-						// maintain task in queue
+					this._forwardItem(element, item, idx);
+
+					renderRun = idx !== this.selected;
```

Comment:
`renderRun` is used for multiple things? This can make two renders occur during the same run?
One thing is to leave the index for resizing next round (which shouldn't be needed if it only applies to selected) and another thing is to decide if we want to render more.

Reply:
If the newly created element is for the currently selected instance we must resize it. That call will be expensive so we keep the current index in the array and next call will resize instance.

Reply:
Ok then I think this would be clearer by just changing the if statement to `if (idx === this.selected)` instead of double negatives and re-using renderRun.
Otherwise you would actually render selected and set renderRun to `false`, causing it to possibly run twice.